test/lit/passes/instrument-memory-filter.wast

```diff
@@ -0,0 +1,552 @@
+  ;; CHECK-NEXT:     (call $load_ptr
+  ;; CHECK-NEXT:      (i32.const 26)
+  ;; CHECK-NEXT:      (i32.const 4)
+  ;; CHECK-NEXT:      (i32.const 11)
+  ;; CHECK-NEXT:      (i32.const 0)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (call $load_val_i64
+  ;; CHECK-NEXT:    (i32.const 27)
+  ;; CHECK-NEXT:    (i64.load offset=12 align=2
+  ;; CHECK-NEXT:     (call $load_ptr
+  ;; CHECK-NEXT:      (i32.const 27)
+  ;; CHECK-NEXT:      (i32.const 8)
+  ;; CHECK-NEXT:      (i32.const 12)
+  ;; CHECK-NEXT:      (i32.const 0)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (call $load_val_f32
+  ;; CHECK-NEXT:    (i32.const 28)
+  ;; CHECK-NEXT:    (f32.load offset=13 align=2
+  ;; CHECK-NEXT:     (call $load_ptr
+  ;; CHECK-NEXT:      (i32.const 28)
+  ;; CHECK-NEXT:      (i32.const 4)
+  ;; CHECK-NEXT:      (i32.const 13)
+  ;; CHECK-NEXT:      (i32.const 0)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT:  (drop
+  ;; CHECK-NEXT:   (call $load_val_f64
+  ;; CHECK-NEXT:    (i32.const 29)
+  ;; CHECK-NEXT:    (f64.load offset=14 align=2
+  ;; CHECK-NEXT:     (call $load_ptr
+  ;; CHECK-NEXT:      (i32.const 29)
+  ;; CHECK-NEXT:      (i32.const 8)
+  ;; CHECK-NEXT:      (i32.const 14)
+  ;; CHECK-NEXT:      (i32.const 0)
+  ;; CHECK-NEXT:     )
+  ;; CHECK-NEXT:    )
+  ;; CHECK-NEXT:   )
+  ;; CHECK-NEXT:  )
+  ;; CHECK-NEXT: )
+  (func $A (type $1)
+    (drop (memory.grow (i32.const 4)))
+
+    (drop (i32.load8_s (i32.const 0)))
+    (drop (i32.load8_u (i32.const 0)))
+    (drop (i32.load16_s (i32.const 0)))
+    (drop (i32.load16_u (i32.const 0)))
+    (drop (i32.load (i32.const 0)))
+    (drop (i64.load8_s (i32.const 0)))
+    (drop (i64.load8_u (i32.const 0)))
+    (drop (i64.load16_s (i32.const 0)))
+    (drop (i64.load16_u (i32.const 0)))
+    (drop (i64.load32_s (i32.const 0)))
+    (drop (i64.load32_u (i32.const 0)))
+    (drop (i64.load (i32.const 0)))
+    (drop (f32.load (i32.const 0)))
+    (drop (f64.load (i32.const 0)))
+
+    (drop (i32.load8_s offset=1 align=1 (i32.const 0)))
+    (drop (i32.load8_u offset=2 align=1 (i32.const 0)))
+    (drop (i32.load16_s offset=3 align=1 (i32.const 0)))
+    (drop (i32.load16_u offset=4 align=1 (i32.const 0)))
+    (drop (i32.load offset=5 align=2 (i32.const 0)))
+    (drop (i64.load8_s offset=6 align=1 (i32.const 0)))
+    (drop (i64.load8_u offset=7 align=1 (i32.const 0)))
+    (drop (i64.load16_s offset=8 align=1 (i32.const 0)))
+    (drop (i64.load16_u offset=9 align=1 (i32.const 0)))
+    (drop (i64.load32_s offset=10 align=2 (i32.const 0)))
+    (drop (i64.load32_u offset=11 align=2 (i32.const 0)))
+    (drop (i64.load offset=12 align=2 (i32.const 0)))
+    (drop (f32.load offset=13 align=2 (i32.const 0)))
+    (drop (f64.load offset=14 align=2 (i32.const 0)))
```

Comment:
Can trim this a little - we don't need every possible variation of loads and stores. Maybe 3 or 4 different ones is enough. Ditto below.